File: app/src/main/java/com/ssttkkl/fgoplanningtool/data/item/ItemsDao.java
package com.ssttkkl.fgoplanningtool.data.item;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface ItemsDao {
    @Query("SELECT * FROM item")
    List<ItemDatabaseEntity> getAllItems();

    @Query("SELECT * FROM item")
    LiveData<List<ItemDatabaseEntity>> getAllItemsAsLiveData();

    @Query("SELECT * FROM item WHERE codename=:codename")
    ItemDatabaseEntity getItemByName(String codename);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void updateItem(ItemDatabaseEntity item);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void updateItems(List<ItemDatabaseEntity> item);
}

File: app/src/main/java/com/ssttkkl/fgoplanningtool/data/item/ItemsDao.kt
package com.ssttkkl.fgoplanningtool.data.item

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface ItemsDao {
    @get:Query("SELECT * FROM Item")
    val all: List<Item>

    @get:Query("SELECT * FROM Item")
    val allAsLiveData: LiveData<List<Item>>

    @Query("SELECT * FROM Item WHERE codename=:codename")
    fun get(codename: String): Item?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun update(item: Collection<Item>)

    @Query("DELETE FROM Item")
    fun clear()
}
